github/client.go

```diff
@@ -215,6 +215,17 @@ func (client *Client) Repository(project *Project) (repo *Repository, err error)
 	return
 }
 
+func (repo *Repository) HasBranch(client *Client, branch string) bool {
+	api, err := client.simpleApi()
+	if err != nil {
+		return false
+	}
+
+	res, err := api.Get(fmt.Sprintf("/repos/%s/%s/branches/%s", repo.Owner.Login, repo.Name, branch))
```

Comment:
Instead of fetching branch info, and thus generating 1 extra API request, how about we simply improve the `invalid value for "head"` error message and replace it with the error message you propose in this PR? That way, the user gets the desired outcome (a more helpful error message), but we don't have to actually change any of the mechanism of opening a PR.

Reply:
I'm happy to change the PR to do that 👍 My thinking was just that this way the user gets an error message before typing a description of the PR (in an enditor not just passing the `-m` argument) while if we change just the error message the user writes a description for a PR that can't be created.

Reply:
Ah I see; thanks for explaining.

Another alternative way of implementing this that avoids an API request would be to check for whether there is a remote tracking branch for the current branch. If there is none, we can assume that the branch wasn't pushed yet. This check should only ever kick in if `-h` *wasn't* passed. Does this sound ok?

Reply:
Sorry for my late reply and thanks for your review! Yes, this sounds good to me. I will try to update the PR.

Reply:
After looking into the code it seems like at the moment we default to the current branch name if there is no head and we don't have a tracking branch:
https://github.com/github/hub/blob/94a6cf45588fb7ae3b695731a518924d75e8348c/commands/pull_request.go#L172-L174

Just wanted to check whether we should still go ahead with this alternative then as it might break people's usage?

Reply:
There are two ways we can check that the current branch was pushed somewhere:
1. the current branch has tracking information: `git config --show-origin --get-regexp 'branch\.BRANCH\.*`; or
2. there is a `.git/refs/remotes/REMOTE/BRANCH` file to indicate that there exists a remote tracking branch.

Of course, it's still possible that the branch was somehow pushed and both checks (1) and (2) still fail, but I think that would be a rare case and the user should be able to work around it by using `-f / --force`.

Does this make sense?

Reply:
Yeah, this makes it clear @mislav , thanks! 👍 I will add those checks instead of the additional API call and assume the branch has not been pushed if both of these fail.